### ISSUE TITLE
AppImage ubuntu 20.04 with or without sudo no HW Acceleration with AMD Vega 64.

### ISSUE THREAD

**Issue by EricLehnen** [mltframework/shotcut#892]
No Hardware Acceleration could be detected. No hardware Video encoding works. Only sound is in the file. No Video.

1. Ubuntu 20.04
2. Shotcut 20.07.11 AppImage
3. AMD RocM Driver installed. -> https://rocmdocs.amd.com/en/latest/Installation_Guide/Installation-Guide.html#ubuntu
4. Vainfo =
vainfo
libva info: VA-API version 1.7.0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/radeonsi_drv_video.so
libva info: Found init function __vaDriverInit_1_7
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.7 (libva 2.6.0)
vainfo: Driver version: Mesa Gallium driver 20.0.8 for Radeon RX Vega (VEGA10, DRM 3.37.0, 5.4.0-42-generic, LLVM 10.0.0)
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            :	VAEntrypointVLD
      VAProfileMPEG2Main              :	VAEntrypointVLD
      VAProfileVC1Simple              :	VAEntrypointVLD
      VAProfileVC1Main                :	VAEntrypointVLD
      VAProfileVC1Advanced            :	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointEncSlice
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointEncSlice
      VAProfileHEVCMain               :	VAEntrypointVLD
      VAProfileHEVCMain               :	VAEntrypointEncSlice
      VAProfileHEVCMain10             :	VAEntrypointVLD
      VAProfileNone                   :	VAEntrypointVideoProc

**Comment by ddennedy** [comment]
This is a drawback of our portable binary builds need to run on older, but VA-API is very version sensitive between the library, user space driver, and kernel module. The combination means it only works on older systems such as 16.04 through 18.04. The need to be compatible with older systems by building on an older OS (Ubuntu 16.04) is higher priority than VA-API working. I will revisit this when we upgrade Qt hopefully for the next version. 
You might have better luck with the flatpak since it is based on newer libs. Or you can look for Debian packages that we are not involved with.
See duplicate https://forum.shotcut.org/t/intel-gpu-vaapi-encoding-acceleration-on-ubuntu-20-04/17769

**Comment by rnz** [comment]
With snap same on radeon 5700xt
```
$ lsb_release -a
LSB Version:	core-11.1.0ubuntu2-noarch:printing-11.1.0ubuntu2-noarch:security-11.1.0ubuntu2-noarch
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04 LTS
Release:	20.04
Codename:	focal

$ snap info shotcut
name:      shotcut
summary:   Free, cross-platform, open source video editor
publisher: Meltytech, LLC (meltytech✓)
store-url: https://snapcraft.io/shotcut
contact:   https://forum.shotcut.org/
license:   unset
description: |
  Shotcut is a free, open source, cross-platform video editor for Windows, Mac and Linux. Major
  features include support for a wide range of formats; no import required meaning native timeline
  editing; Blackmagic Design support for input and preview monitoring; and resolution support to 4k.
commands:
  - shotcut.ffmpeg
  - shotcut.ffplay
  - shotcut.ffprobe
  - shotcut.melt
  - shotcut.qmelt
  - shotcut
snap-id:      QbcwWzVel4jXFaSClk6oiPpN3GCTXwvi
tracking:     latest/stable
refresh-date: today at 17:43 MSK
channels:
  latest/stable:    20.07.11 2020-07-11 (68) 90MB classic
  latest/candidate: 20.07.11 2020-07-11 (68) 90MB classic
  latest/beta:      20.07.11 2020-07-11 (68) 90MB classic
  latest/edge:      20.07.11 2020-07-11 (68) 90MB classic
installed:          20.07.11            (68) 90MB classic

$ vainfo
libva info: VA-API version 1.7.0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/radeonsi_drv_video.so
libva info: Found init function __vaDriverInit_1_7
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.7 (libva 2.6.0)
vainfo: Driver version: Mesa Gallium driver 20.0.8 for AMD Radeon RX 5700 XT (NAVI10, DRM 3.36.0, 5.6.0-1020-oem, LLVM 10.0.0)
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            :	VAEntrypointVLD
      VAProfileMPEG2Main              :	VAEntrypointVLD
      VAProfileVC1Simple              :	VAEntrypointVLD
      VAProfileVC1Main                :	VAEntrypointVLD
      VAProfileVC1Advanced            :	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointEncSlice
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointEncSlice
      VAProfileHEVCMain               :	VAEntrypointVLD
      VAProfileHEVCMain               :	VAEntrypointEncSlice
      VAProfileHEVCMain10             :	VAEntrypointVLD
      VAProfileJPEGBaseline           :	VAEntrypointVLD
      VAProfileVP9Profile0            :	VAEntrypointVLD
      VAProfileVP9Profile2            :	VAEntrypointVLD
      VAProfileNone                   :	VAEntrypointVideoProc
```
![image](https://user-images.githubusercontent.com/851927/88482889-d4b0de00-cf6c-11ea-9bed-f27d4808e53b.png)
![image](https://user-images.githubusercontent.com/851927/88482894-e2666380-cf6c-11ea-9925-c8822adec5e4.png)

Result: encoding to fast, and no video in exported file

**Comment by ddennedy** [comment]
The snap is the same build, slightly different bundling, as the AppImage and portable tar. The Flatpak is a completely different build, however, and VAAPI works there for me on Ubuntu 20.04.

**Comment by alexmyczko** [comment]
What about the official deb package?

**Comment by ddennedy** [comment]
Even with the new beta 20.10.18 based on Ubuntu 18.04, VA-API does not work on Ubuntu 20.04. VA-API, I have concluded, is completely opposed to the concept of portable binary apps, and there is nothing I can do that I know of to fix their rigid version dependencies between app, library, userspace drivers, and kernel module. Workarounds are to use Flatpak or a distro package if you really need to use VA-API. Honestly, the only decent hardware encoding is NVIDIA NVENC with Turing architecture (e.g. RTX 20/30xx) or later. Closing as won't fix.
